osv/models.py

```diff
@@ -352,9 +352,19 @@ def _pre_put_hook(self):
     self.ecosystem = list(ecosystems_set)
     self.ecosystem.sort()
 
-    self.purl = list({
-        pkg.package.purl for pkg in self.affected_packages if pkg.package.purl
-    })
+    def with_and_without_qualifiers(affected_packages) -> list[str]:
+      resulting_set = set()
```

Comment:
Does this need to be a set at all?

Reply:
Not sure, it was a set in the previous implementation, so I just added it here. I'm guessing it would only matter if two affected packages have the same purl

Reply:
Ah, ack. This should indeed still be a set.